### ISSUE TITLE
Songs are skipping on Spotify

### ISSUE THREAD

**Issue by AryToNeX** [Eloston/ungoogled-chromium#1258]
**Describe the bug**
Songs are skipping on Spotify. Advertisements play just fine. Sometimes a song seems to play fine but stops at the ninth second of playback.

**To Reproduce**
1. Install Widevine as per the FAQ
2. Go to 'open.spotify.com'
3. Log in
4. Try to play a song
5. "Hear" the bug

**Expected behavior**
The song should play just fine.

**Environment (please complete the following information):**
 - OS/Platform and version: Windows 10 20H1
 - ungoogled-chromium version: 86.0.4240.183

**Comment by mmshivesh** [comment]
Have the same issue

**Comment by jgarciadcruz** [comment]
+1 on this issue. Windows 10, running **87.0.4280.67** (installer version).

**Comment by r3mis** [comment]
Same. Tested on `87.0.4280.88`. 
It will skip 3-4 songs and stop reproducing music.
Looks like, after 10 seconds, audio will even stop working.

**Comment by EPIVISION** [comment]
can confirm on _ungoogled-chromium_87.0.4280.141-1.1_windows-x64_

also tried multiple WidevineCdm versions:
4.10.1582.2 (current dl.google.com/widevine-cdm) ❌ 
4.10.1679.0 (bundled in chromium-dev#90.0.4399.0-r846566) ❌

**Comment by EPIVISION** [comment]
88.0 same issue.
any further insight?

**Comment by vairag22** [comment]
Using `Chromium 89.0.4389.90` `Widevine 4.10.2209.0` on Windows 10 x64
This issue is still not fixed. I wonder if this is a Windows only issue or it happens on Linux and Mac as well?

**Comment by mmshivesh** [comment]
> > Using `Chromium 89.0.4389.90` `Widevine 4.10.2209.0` on Windows 10 x64
> > This issue is still not fixed. I wonder if this is a Windows only issue or it happens on Linux and Mac as well?
> 
> Windows only, it seems. On my Linux machines, Widevine works fine.

I can confirm it still happens on my Mac with Version `88.0.4324.192 `. The Widevine module however seems to be working fine though, that's what makes this whole thing surprising (I tested it [here](https://bitmovin.com/demos/drm)).

However, this issue doesn't happen on Chromium Edge which points to it being something related to Ungoogled Chromium.

**Comment by vairag22** [comment]
@AryToNeX @mmshivesh Yep. I also did my testing and found the same thing.

Spotify web player works alright on my Debian machine but not on my Windows 10 laptop. I don't have a Mac so I am taking Shivesh's word on it that it does not work on MacOS either. Anyways this does not seem to be a Widevine or missing component issue coz all the tests given [here](https://chromium.woolyss.com/#widevine) returned positive. This maybe bug related to **ungoogled-chromium** because Spotify web player worked fine for me on **Edge** (based on Chromium 89) and **Brave** (didn't test, but I used Brave last summer and it worked fine), though few users have reported facing such issues. Thanks to @woolyss, with whom I've discussing this issue, I got this [screenshot](https://user-images.githubusercontent.com/71322586/113344540-d417e500-934e-11eb-815c-ad94053f3b2a.jpg) (POST requests are forbidden, error 403) and links to some pretty interesting discussions from [Spotify forum](https://community.spotify.com/t5/Other-Partners-Web-Player-etc/Web-player-skips-tracks-or-freezes-after-9-seconds/td-p/5079716) (check the console log screenshots and the last comment) and [Vivaldi forum](https://forum.vivaldi.net/topic/51775/spotify-web-player-broken). 

My guess is that this bug arised when @Eloston was skimming Chromium to "ungoogle" it and the browsers which share the same codebase or did similar thing have this bug. It seems Vivaldi has resolved this issue but I haven't tested it yet, it would very interesting to see what fixing Vivaldi folks have done.

Ran my tests on `Chromium (ungoogled) 89.0.4389.90 64-bit + Widevine 4.10.2209.0 + Proprietary codecs`, [Windows](https://github.com/macchrome/winchrome/releases/download/v89.0.4389.114-r843830-Win64/ungoogled-chromium-89.0.4389.114-1_Win64.7z) and [Linux](https://github.com/macchrome/linchrome/releases/download/v89.0.4389.90-r843830-portable-ungoogled-Lin64/ungoogled-chromium_89.0.4389.90_1.vaapi_linux.tar.xz) builds provided by Marmaduke (@macchrome).

**Comment by Eloston** [comment]
Are the requests that are failing on ungoogled-chromium present in the other browsers? Are they succeeding on the other browsers? Can we compare the request payloads of such requests?

**Comment by vairag22** [comment]
> Are the requests that are failing on ungoogled-chromium present in the other browsers? Are they succeeding on the other browsers?
POST requests are failing only on ungoogled-chromium. They succeed on other browsers.

**Comment by vairag22** [comment]
> Are the requests that are failing on ungoogled-chromium present in the other browsers? Are they succeeding on the other browsers?

POST requests are failing only on ungoogled-chromium. They succeed on other browsers.

> Can we compare the request payloads of such requests?

maybe 😕

**Comment by mmshivesh** [comment]
> Are the requests that are failing on ungoogled-chromium present in the other browsers? Are they succeeding on the other browsers? Can we compare the request payloads of such requests?

I'd be happy to provide any help if needed

**Comment by vairag22** [comment]
_> updated to version 90.0.4430.72_
_> the saga continues..._

**Comment by Eloston** [comment]
@mmshivesh Based on what @vairag22 found, we probably want to investigate the payloads of those POST requests to see what's different. I don't have a Spotify account, but if someone has a way I can test as an anonymous user, I could take a look.

**Comment by r3mis** [comment]
You could try to make an "anonymous" account with a temporally E-mail. No?

**Comment by woolyss** [comment]
> You could try to make an "anonymous" account with a temporally E-mail. No?
No temporary email allowed on Spotify.

Note an important thing about Spotify : It needs support of the H.264 codec + Widevine CDM plugin.

Hibbiki, a developer, explains me that ever since Widevine L3 been publicly broken, Google has tighten out security around it and pretty much all resources using Widevine now require VMP (Verified Media Path) on Windows and Mac to be enabled.

Official explanation at https://www.widevine.com/news (Browser CDM - Verified Media Path)

The process for VMP certification is to simply generate signature (for Chromium that would be both chrome.exe and chrome.dll) signed with appropriate Widevine-accepted certificate (note this is different from signing binary on Windows and if both are done, VMP must be performed after).

I know few open-source browsers like Brave and Vivaldi work nicely with the Spotify website. I tested them.

Their official app works nicely.
https://www.spotify.com/us/download/other/

**Comment by r3mis** [comment]
Damn. That sucks. Well, thank you for explaining things here.

**Comment by Eloston** [comment]
@woolyss Out of curiosity, do you know if the VMP check is done asynchronously? It would explain why Spotify suddenly stops working after a few seconds.

I also wonder what it takes to get VMP-certified since Brave and Vivaldi did it. Perhaps they have the resources to do it since they're companies (if I'm not mistaken)?

Also based on a quick search, there seems to be alternative FOSS clients and libraries like [spotify-qt](https://github.com/kraxarn/spotify-qt), [spotifyd](https://github.com/Spotifyd/spotifyd), and [librespot](https://github.com/librespot-org/librespot). Might be worth a shot if the proprietary Widevine CDM plugin isn't to your liking.

**Comment by AryToNeX** [comment]
> Also based on a quick search, there seems to be alternative FOSS clients and libraries like [spotify-qt](https://github.com/kraxarn/spotify-qt), [spotifyd](https://github.com/Spotifyd/spotifyd), and [librespot](https://github.com/librespot-org/librespot). Might be worth a shot if the proprietary Widevine CDM plugin isn't to your liking.

Those libraries only work with Spotify Premium subscriptions, though

**Comment by kaimyel** [comment]
Any developments on this, lad?

**Comment by wchen342** [comment]
Yep, you need to be a company to get your key signed. See https://github.com/brave/browser-laptop/issues/10449. Also Google is explicitly hostile to open source projects on this, see https://blog.samuelmaddock.com/widevine/gmail-thread.html and https://github.com/electron/electron/issues/12427.

**Comment by vairag22** [comment]
> Yep, you need to be a company to get your key signed. See [brave/browser-laptop#10449](https://github.com/brave/browser-laptop/issues/10449). Also Google is explicitly hostile to open source projects on this, see https://blog.samuelmaddock.com/widevine/gmail-thread.html and [electron/electron#12427](https://github.com/electron/electron/issues/12427).

Hmm... looks like nothing could be done about this issue. Just hope they don't figure out a way to spread this plague on Linux.

Anyway, how did Brave and Firefox manage to get their licenses? They don't qualify as open-source projects? I guess as long as you pawn your users with open-source software or write de-platforming articles instead of code, Google will send you VMP certificates with chocolates and roses.

**Comment by wchen342** [comment]
Firefox is a big enough project and Mozilla is behind it. Brave is not as big but they are a company and have more leverage to afford the back and forth with Google. But for smaller projects and individuals it won't be the case.

> I guess as long as you pawn your users with open-source software or write de-platforming articles instead of code, Google will send you VMP certificates with chocolates and roses.

That is most likely not true. I think they are strict on this because big corps see Linux users and devs as a risk when it comes to DRM (which is not necessarily true but that's the myth).

**Comment by vairag22** [comment]
>> I guess as long as you pawn your users with open-source software or write de-platforming articles instead of code, Google will send you VMP certificates with chocolates and roses.
>
> Firefox is a big enough project and Mozilla is behind it. Brave is not as big but they are a company and have more leverage to afford the back and forth with Google. But for smaller projects and individuals it won't be the case.

Yeah I know. The "chocolate and roses" case is extremely unlikely but that wasn't my point. Google clearly said they don't _[plan to]_ support open-source projects in [this email conversation](https://blog.samuelmaddock.com/widevine/gmail-thread.html). If that's really the case then they shouldn't make any exceptions, otherwise what should I call it except double-standards and big tech bias?

**Comment by vairag22** [comment]
Does Spotify, Netflix, etc. work as expected if someone builds Brave or Firefox on Windows/Mac by themselves?

**Comment by mohitmujawdiya** [comment]
This issue still exists. Will it be resolved?

**Comment by tylerrobb** [comment]
Still experiencing this issue on the Spotify web player after installing Widevine (win_x64 - 4.10.2391.0)

**Comment by KHR3b** [comment]
Same, also doesn't work on Firefox ESR

**Comment by amrbadreldin** [comment]
still facing the issue of Spotify skipping songs in ungoogled chromium (using Marmaduke (@macchrome) UC Builds because it patches a lot of things yet that Spotify issue still remains.)

**Comment by networkException** [comment]
I don't know what to tell you, it has already been answered that we are unlikely to support this (basically ever): https://github.com/ungoogled-software/ungoogled-chromium/issues/1258#issuecomment-873456050

**Comment by mohitmujawdiya** [comment]
<details>
  <summary>If you think piracy is fine</summary>
  
An alternative I am using for more than a year is cracked Spotify with ads blocked. You can download it via torrents and it's smooth as heck. No problems whatsoever, you can also make the "updates" folder read-only so that it can't auto-update.
  
</details>

**Comment by GunGunGun** [comment]
Thank to Google the web has become such a bs place.

**Comment by ygxrr** [comment]
can confirm this issue is still present on version 127.0.6533.119 but only in Windows, on my netbook running Void Linux Spotify works fine on ungoogled-chromium
